sdk/tdf.go

```diff
@@ -392,12 +442,21 @@ func (s SDK) LoadTDF(reader io.ReadSeeker) (*Reader, error) {
 	}, nil
 }
 
+// Do any network based operations required.
+// This allows making the requests cancellable
+func (r *Reader) Init(ctx context.Context) error {
+	if r.payloadKey != nil {
+		return nil
+	}
+	return r.doPayloadKeyUnwrap(ctx)
+}
+
 // Read reads up to len(p) bytes into p. It returns the number of bytes
 // read (0 <= n <= len(p)) and any error encountered. It returns an
 // io.EOF error when the stream ends.
 func (r *Reader) Read(p []byte) (int, error) {
```

Comment:
I wonder if we should break the contract and pass in context

Reply:
yeah, I kinda wish there was a 'ReadWithContext' interface. It is very common pattern for read to use network resources, either explicitly (e.g. using an HTTP request with a Range header, like we do for remote decrypt), or implicitly using something like NFS. I'd rather leave this as is for now and see how people use it to decide to change it